src/domain/collection/curatedEventList/__tests__/CuratedEventList.test.tsx

```diff
@@ -104,3 +149,40 @@ test('should show expired events', async () => {
     expect(screen.getByText(eventName)).toBeInTheDocument();
   });
 });
+
+test('event list pagination works', async () => {
+  advanceTo('2020-10-05');
+  const eventsCount = 35;
+  const { collection, mocks, eventNames } = getMocksForPagination(eventsCount);
+  render(<CuratedEventList collection={collection} />, {
+    mocks,
+  });
+  await waitForRequestToComplete();
+
+  const clickShowMoreEventsButton = (name: RegExp) => {
+    userEvent.click(screen.getByRole('button', { name, hidden: true }));
+  };
+
+  // use loop to fetch all the events by clicking show more button (pagination)
+  for (
+    let eventsFetchedCount = PAGE_SIZE;
+    eventsFetchedCount < eventsCount;
+    eventsFetchedCount += PAGE_SIZE
+  ) {
+    clickShowMoreEventsButton(
+      new RegExp(
+        `Näytä lisää tapahtumia \\(${eventsCount - eventsFetchedCount}\\)`,
+        'i'
+      )
+    );
+    await waitForRequestToComplete();
+  }
+
+  expect(
+    screen.queryByRole('button', { name: /näytä lisää tapahtumia/ })
+  ).not.toBeInTheDocument();
+
+  for (const eventName of eventNames) {
```

Comment:
It seems that we keep on clicking the showMorebutton until it vanishes and then check that all events are present. What about checking the event names that belongs to particular chunk, that they are present (inside the loop)?
Perhaps it does not matter except if events would somehow would come in unexpected order, then this would not catch it

Reply:
Added tests to the pagination loop